pex/bin/pex.py

```diff
@@ -647,8 +658,12 @@ def walk_and_do(fn, src_dir):
       for resolved_dist in resolveds:
         log('  %s -> %s' % (resolved_dist.requirement, resolved_dist.distribution),
             V=options.verbosity)
-        pex_builder.add_distribution(resolved_dist.distribution)
-        pex_builder.add_requirement(resolved_dist.requirement)
+
+        if options.dehydrated:
+          pex_info.add_dehydrated_requirement(resolved_dist.distribution.as_requirement())
```

Comment:
Why is a new field in PEX-INFO needed at all. Can't the existing requirements be used and just addition of distributions be skipped? IE: On pex boot, if len(requirements) > 0 and len(distributions) == 0, we're in rehydration mode. Further, this would imply that re-hydration logic belongs in PEXEnvironment, where distributions are normally found inside the PEX and added to the sys.path. Inventing a new way to add these to the sys.path below worryingly expands the testing parameter space for no good reason I can see.